draft-pda-protocol.md

```diff
@@ -146,13 +146,164 @@ document.
+| Client +---------->   Leader   +---------> Collector |
+|        |    +----->            |         |           |
++--------+    |     +-----^------+         +-----------+
+              |           |
++--------+    |           |
+|        |    |           |
+| Client +----+     +-----V------+
+|        |          |            |
++--------+          |   Helper   |
+                    |            |
+                    +------------+
+~~~~
+{: #pa-topology title="System Architecture"}
+
+
+The main participants in the protocol are as follows:
+
+Collector:
+: The entity which wants to take the measurement and ultimately receives
+  the results. Any given measurement will have a single collector.
+
+Client(s):
+: The endpoints which directly take the measurement(s) and report it to the
+  PPM system. In order to provide reasonable levels of privacy, there
+  must be a large number of clients.
+
+Aggregator:
+: An endpoint which receives report shares. Each aggregator works with the
+  other aggregators to compute the final aggregate. This protocol defines
+  two types of aggregators: a Leader and a set of Helpers.
+
+Leader:
+: The leader is responsible for coordinating the protocol. It receives
+  the encrypted shares, distributes them to the helpers and orchestrates
+  the process of computing the final measurement.
+
+Helper:
+: Helpers are responsible for executing the protocol as instructed
+  by the leader. The protocol is designed so that helpers can be relatively
+  lightweight, with most of the state held at the leader.
+{:br}
+
+
+In order to take a measurement, the collector first decides on the
+measurement parameters, including:
+
+* The values to be measured
+* The statistic to be computed (e.g., sum, mean, etc.)
+* The PPM scheme to use. This is to some extent dictated by the previous
+  choices.
+
+These parameters are distributed out of band to the clients and to
+the aggregators.
+
+During the duration of the measurement, each client records its own
+value(s), packages them up into a report and sends them to the leader.
+Each share is separately encrypted for each aggregator so that even
+though they pass through the leader, it is unable to see or modify
+them. Depending on the measurement, the client may only send one
+report or may send many reports over time.
+
+The leader distributes the shares to the helpers and orchestrates
+the process of verifying them (see {{validating-inputs}})
+and assembling them into a final measurement for the collector.
+Depending on the PPM scheme, it may be possible to incrementally
+process each report as it comes in, or may be necesary to wait
+until all the required shares are received. This protocol is
+compatible with both a "push" mode in which the leader computes
+the measurements at predetermined points and sends the results
+to the collector and a "pull" mode in which the collector tells
+the leader when to perform the aggregation and return the results.
```

Comment:
```suggestion
until all the required shares are received. 

This protocol is compatible with both a "push" deployment mode in which the 
leader computes the measurements at predetermined points and sends the results
to the collector and a "pull" mode in which the collector tells
the leader when to perform the aggregation and return the results.
```

Reply:
Tried to clarify that modes are a deployment choice, and not something baked into the protocol.